src/h3lib/lib/vertex.c

```diff
@@ -269,3 +269,26 @@ void H3_EXPORT(vertexToPoint)(H3Index vertex, GeoCoord* coord) {
     // Copy from boundary to output coord
     *coord = gb.verts[0];
 }
+
+/**
+ * Whether the input is a valid H3 vertex
+ * @param  vertex H3 index possibly describing a vertex
+ * @return        Whether the input is valid
+ */
+int H3_EXPORT(isValidVertex)(H3Index vertex) {
```

Comment:
Do vertices have a unique representation? That is, for a given vertex, is it only allowed a single cell "owner"?

It doesn't look like this code checks if the vertex has the correct owner.

Reply:
My assumption is this function will assure the vertex is in the canonical form, if we want to support non-canonical vertexes, perhaps we need `canonicalizeVertex`/`isCanonicalVertex`?

Reply:
Hm. That's a good point, though there's no way at present that you can create a non-canonical vertex, so I'm leery about adding public functions addressing a practically unreachable issue. If we wanted this check, I think I'd need to implement `vertexToCells` first, and then check that the owner had the lowest numerical index. 

I don't have a clear picture for the validation functions on whether we should prioritize speed (because you might need to validate many indexes at once) or exhaustiveness, which in this case would mean an expensive check for an unlikely case.

So I guess this brings up two questions:
- Is there a use for a non-canonical vertex (i.e. an index indicating a cell and vertex number)? This would be easy to produce, but I'm not sure it has any value, and it would muddy the waters about whether all vertexes are canonical.
- If not, is it worth an expensive check in this function for a case that at present would need to be hand-crafted?

Reply:
I think we definitely want uniqueness for the indices (and thus not allow "non-canonical" vertices), and if we have a function called `isValidVertex` it has to be exhaustive and strictly correct. I think speed is secondary to correctness, and is something we could potentially improve in the future.

Reply:
Updated to check validity by recreating the valid vertex and comparing, plus added a test for the incorrect-owner case. Recreating the vertex also covers the invalid vertex number case, so this simplified things nicely.